### PR TITLE
update test peer bmc communication

### PR DESCRIPTION
**Pull request by WangFanta**

### CODE REVIEW

File: test/functional/test_bmc_peer_communication.py [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    fi.close()
+    return fi.name
+
+
+def setup_module():
+    global ivn_file
+    os.environ['PATH'] = new_path
+    ivn_file = saved_config_file()
+
+    global topo
```
Comment (weia5-emc) [inline test/functional/test_bmc_peer_communication.py]: can be removed.

File: test/functional/test_bmc_peer_communication.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        global emu_file_for_del1
+        os.system("sed -i '$d' {}".format(emu_file_for_del0))
+        os.system("sed -i '$d' {}".format(emu_file_for_del0))
+        os.system("sed -i '$d' {}".format(emu_file_for_del1))
+        os.system("sed -i '$d' {}".format(emu_file_for_del1))
+        test_bmc_communication._stop_node(fake_node0)
+        test_bmc_communication._stop_node(fake_node1)
+        fake_node0 = None
+        fake_node1 = None
+        topo.delete()
```
Comment (weia5-emc) [inline test/functional/test_bmc_peer_communication.py]: don't need it here

File: test/functional/test_bmc_peer_communication.py [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+        emu_file_for_del1 = fake_node1_list[1]
+
+    @classmethod
+    def tearDownClass(cls):
+        global fake_node0
+        global fake_node1
+        global topo
+        global emu_file_for_del0
+        global emu_file_for_del1
+        os.system("sed -i '$d' {}".format(emu_file_for_del0))
```
Comment (weia5-emc) [inline test/functional/test_bmc_peer_communication.py]: don't need it. 
terminate_workspace() will delete it.

File: test/functional/test_bmc_peer_communication.py [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+        fake_node_obj.init()
+        fake_node_obj.precheck()
+        fake_node_obj.start()
+        return fake_node_obj
+
+    @staticmethod
+    def node_cfg_up(node_name, ns_name, boot_image, node_type):
+        global add_content0
+        global add_content1
+        global emu_file_for_del0
```
Comment (weia5-emc) [inline test/functional/test_bmc_peer_communication.py]: useless line can be removed.

File: test/functional/test_bmc_peer_communication.py [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+    def client_ssh(self, ns_ip):
+        ssh = sshclient.SSH(host=ns_ip, username="root", password="root", port=8022)
+        ssh.wait_for_host_up()
+        return ssh
+
+    def test_peer_bmc_fru(self):
+        global cmd_prefix0
+        global cmd_prefix1
+        global fake_node0
```
Comment (weia5-emc) [inline test/functional/test_bmc_peer_communication.py]: useless code can be removed.

File: test/functional/test_bmc_peer_communication.py [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+        status, result0 = node0_ssh.exec_command(cmd_prefix0[0])
+        node0_ssh = self.client_ssh('192.168.188.91')
+        status, result1 = node0_ssh.exec_command(cmd_prefix1[0])
+        self.assertIn("PowerEdge R730", result0, "peer_bmc can not Obtain fru")
+        self.assertIn("Quanta Computer", result1, "peer_bmc can not Obtain fru")
+
+    def test_peer_bmc_lan(self):
+        global cmd_prefix0
+        global cmd_prefix1
+        global fake_node0
```
Comment (weia5-emc) [inline test/functional/test_bmc_peer_communication.py]: useless code can be removed.

File: test/functional/test_bmc_peer_communication.py [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+        node0_ssh = self.client_ssh('192.168.188.91')
+        status, result1 = node0_ssh.exec_command(cmd_prefix1[1])
+        self.assertIn("MAC Address", result0, "peer_bmc can not Obtain lan")
+        self.assertIn("MAC Address", result1, "peer_bmc can not Obtain lan")
+        self.assertNotEqual(result0, result1, "peer_bmc can not Obtain lan")
+
+    def test_peer_bmc_sensor(self):
+        global cmd_prefix0
+        global cmd_prefix1
+        global fake_node0
```
Comment (weia5-emc) [inline test/functional/test_bmc_peer_communication.py]: useless code can be removed.

File: test/functional/test_bmc_peer_communication.py [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+        node0_ssh = self.client_ssh('192.168.188.91')
+        status, result0 = node0_ssh.exec_command(cmd_prefix0[2])
+        node0_ssh = self.client_ssh('192.168.188.91')
+        status, result1 = node0_ssh.exec_command(cmd_prefix1[2])
+        self.assertNotEqual(result0, result1, "peer_bmc can not Obtain sensor")
+
+    def test_peer_bmc_sel(self):
+        global cmd_prefix0
+        global cmd_prefix1
+        global fake_node0
```
Comment (weia5-emc) [inline test/functional/test_bmc_peer_communication.py]: useless code can be removed.

File: test/functional/test_bmc_peer_communication.py [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+        global fake_node1
+        node0_ssh = self.client_ssh('192.168.188.91')
+        status, result0 = node0_ssh.exec_command(cmd_prefix0[3])
+        node0_ssh = self.client_ssh('192.168.188.91')
+        status, result1 = node0_ssh.exec_command(cmd_prefix1[3])
+        self.assertNotEqual(result0, result1, "peer_bmc can not Obtain sensor")
+
+    def test_peer_node_off(self):
+        global cmd_prefix0
```
Comment (weia5-emc) [inline test/functional/test_bmc_peer_communication.py]: 4 lines are useless can be removed.

File: test/functional/test_bmc_peer_communication.py [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+        global cmd_prefix1
+        global fake_node0
+        global fake_node1
+        os.system(cmd_prefix1[4])
+        PS_QEMU = "ps ax | grep qemu"
+        qemu_result = run_command(PS_QEMU, True, subprocess.PIPE, subprocess.PIPE)[1]
+        self.assertNotIn("test1", qemu_result, "peer_bmc can not power off")
+
+    def test_peer_node_on(self):
+        global cmd_prefix0
```
Comment (weia5-emc) [inline test/functional/test_bmc_peer_communication.py]: 4 lines are useless can be removed.

File: test/functional/test_bmc_peer_communication.py [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+        global cmd_prefix1
+        global fake_node0
+        global fake_node1
+        os.system(cmd_prefix1[5])
+        node1_ssh = self.client_ssh('192.168.188.92')
+        status, result = node1_ssh.exec_command('ls')
+        assert result
+
+    def test_peer_node_cycle(self):
+        global cmd_prefix0
```
Comment (weia5-emc) [inline test/functional/test_bmc_peer_communication.py]: 4 lines are useless can be removed.

File: test/functional/test_bmc_peer_communication.py [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+    def node_cfg_up(node_name, ns_name, boot_image, node_type):
+        global add_content0
+        global add_content1
+        global emu_file_for_del0
+        global emu_file_for_del1
+        fake_node = None
+        fake_node = copy.deepcopy(FakeConfig().get_node_info())
+        emu_dir = os.path.join(config.infrasim_data, node_type)
+        emu_file = os.path.join(emu_dir, node_type + ".emu")
+        if ns_name == "node0ns":
```
Comment (weia5-emc) [inline test/functional/test_bmc_peer_communication.py]: 👏

**Comment by weia5-emc** [comment]
👌

**Comment by chenge3** [comment]
👍

**Pull request merged**